client/components/AIAssistant.js

```diff
@@ -38,17 +45,62 @@ export const AIAssistant = () => {
         date: new Date(),
       },
     ]);
-    sendAppMessage(
-      {
-        kind: "assist",
-        query,
-      },
-      "*",
-    );
+    try {
+      setIsPrompting(true);
+      const response = await fetchQuery(roomUrl, query);
+      setChatHistory((prev) => [
+        ...prev,
+        {
+          role: "assistant",
+          content: response,
+          date: new Date(),
+        },
+      ]);
+      playAudioMsg();
+    } catch {
+      setChatHistory((prev) => [
+        ...prev,
+        {
+          role: "assistant",
+          content:
+            "Uh oh! While I tried to get a response for you, an error occurred! Please try again.",
+          date: new Date(),
+        },
+      ]);
+      playAudioError();
+    } finally {
+      setIsPrompting(false);
+    }
   };
 
-
-  const chatRef = useRef(null);
+  const handleSummaryClick = async () => {
+    try {
+      setIsSummarizing(true);
+      const response = await fetchSummary(roomUrl);
+      setChatHistory((prev) => [
+        ...prev,
+        {
+          role: "assistant",
+          content: response,
+          date: new Date(),
+        },
+      ]);
+      playAudioMsg();
+    } catch {
+      setChatHistory((prev) => [
```

Comment:
We seem to have two of the same chunks of code here and above - is it worth extracting them into a separate function for reuse?

Reply:
The functions are not _exactly_ the same, but I can see a net win in exporting the message object creation into functions.